File: common/data_loader.js
const fs = require('fs');
const path = require('path');

function loadData(dataPath) {
    const data = [];
    const stats = fs.statSync(dataPath);

    if (stats.isDirectory()) {
        const files = fs.readdirSync(dataPath);
        for (const file of files) {
            data.push(fs.readFileSync(path.join(dataPath, file), 'utf8'));
        }
    } else if (dataPath.endsWith('.jsonl')) {
        const lines = fs.readFileSync(dataPath, 'utf8').split('\n').filter(Boolean);
        for (const line of lines) {
            data.push(JSON.parse(line).payload);
        }
    } else if (dataPath.endsWith('.csv')) {
        const lines = fs.readFileSync(dataPath, 'utf8').split('\n').filter(Boolean);
        for (let i = 1; i < lines.length; i++) {
            data.push(lines[i]);
        }
    }
    return data;
}

module.exports = { loadData };

File: csharp/DataLoader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

namespace Common
{
    public static class DataLoader
    {
        public static List<string> LoadData(string path)
        {
            var data = new List<string>();

            if (Directory.Exists(path))
            {
                foreach (var file in Directory.GetFiles(path))
                {
                    data.Add(File.ReadAllText(file));
                }
            }
            else if (path.EndsWith(".jsonl"))
            {
                foreach (var line in File.ReadLines(path))
                {
                    using (JsonDocument doc = JsonDocument.Parse(line))
                    {
                        data.Add(doc.RootElement.GetProperty("payload").GetString());
                    }
                }
            }
            else if (path.EndsWith(".csv"))
            {
                var lines = File.ReadAllLines(path);
                for (int i = 1; i < lines.Length; i++) // skip header
                {
                    data.Add(lines[i]);
                }
            }

            return data;
        }
    }
}
